pkg/handler/common/provider/alibaba.go

```diff
@@ -283,3 +283,27 @@ func AlibabaVswitchesWithClusterCredentialsEndpoint(ctx context.Context, userInf
 
 	return ListAlibabaVSwitches(accessKeyID, accessKeySecret, region)
 }
+
+func getAlibabaClient(accessKeyID, accessKeySecret, region string) (*ecs.Client, error) {
+	client, err := ecs.NewClientWithAccessKey(region, accessKeyID, accessKeySecret)
+	if err != nil {
+		return nil, utilerrors.New(http.StatusInternalServerError, fmt.Sprintf("failed to create client: %v", err))
+	}
+	return client, err
+}
+
+func AlibabaInstanceTypes(accessKeyID, accessKeySecret, region, instanceType string) (*ecs.DescribeInstanceTypesResponse, error) {
```

Comment:
why not make this GetAlibabaInstanceType(or similar name), and just get that type, so that in [providers.go](https://github.com/kubermatic/kubermatic/pull/10231/files#diff-e2c2e4a1524233a7ee0eea61ef6521948ccd95e38d132b197b2347db11c06d01R531) you dont have to make a loop?


Similarly for DigitalOcean